### PR TITLE
Replace String? = null by `String = ""`

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
Since it's a default value, there seems to be no reason to accept a nullable type here.

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/CoroutineHelpers.kt [hunk 1]
```diff
@@ -45,10 +45,10 @@ import kotlin.coroutines.suspendCoroutine
  *   a non-null localized message.
  */
 suspend fun <T> FragmentActivity.runCatchingTask(
-    errorMessage: String? = null,
+    errorMessage: String = "",
     block: suspend () -> T?
 ): T? {
-    val extraInfo = errorMessage ?: ""
+    val extraInfo = errorMessage
```
Comment (BrayanDSO) [inline AnkiDroid/src/main/java/com/ichi2/anki/CoroutineHelpers.kt:51]: I believe that we can drop the `extraInfo` variable and use errorMessage directly now (extraInfo is used on one of the methods below as well)

**Review by BrayanDSO — APPROVED** [review]
As far as the PR scope goes, LGTM. Left a cleanup idea to do. Implementer's choice

**Review by BrayanDSO — APPROVED** [review]

File: AnkiDroid/src/main/java/com/ichi2/anki/CoroutineHelpers.kt (hunk 1 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/anki/CoroutineHelpers.kt:51]: I agree. I thought I removed it everywhere, strange

**Comment by oakkitten** [comment]
I think `null` better expresses the idea. `null` says that we don't want an accompanying message, and `""` says that we want a message that is empty. I'm not exactly sure what `Timber` will do with that, probably it won't print an extra blank line, but having `null` doesn't make me guess.

**Comment by BrayanDSO** [comment]
I agree with oakkitten's point and `Timber.e` accepts null, so null as default value should be better

**Comment by sumanabhi** [comment]
> I think `null` better expresses the idea. `null` says that we don't want an accompanying message, and `""` says that we want a message that is empty. I'm not exactly sure what `Timber` will do with that, probably it won't print an extra blank line, but having `null` doesn't make me guess.

No Doubt @oakkitten explanation is correct for various scenarios and code standardisation. 

Also, I want to suggest that whenever we are defining `""` at least in the kotlin file we can directly use the `String.empty`
It will make the code more readable.

**Comment by Arthur-Milchior** [comment]
Since Timber can take null, i changed this PR to keep null and not replacing it with ""

**Review by lukstbit — APPROVED** [review]

**Pull request merged**